lib/router/route-info.ts

```diff
@@ -49,15 +49,20 @@ export interface RouteInfo {
   ): RouteInfo | undefined;
 }
 
+export interface RouteInfoWithAttributes extends RouteInfo {
+  attributes: Dict<unknown>;
+}
+
 let ROUTE_INFOS = new WeakMap<InternalRouteInfo<Route>, RouteInfo>();
 
 export function toReadOnlyRouteInfo(
   routeInfos: InternalRouteInfo<Route>[],
-  queryParams: Dict<unknown> = {}
+  queryParams: Dict<unknown> = {},
+  includeAttributes = false
 ) {
   return routeInfos.map((info, i) => {
-    let { name, params, paramNames } = info;
-    let publicRouteInfo = new class implements RouteInfo {
+    let { name, params, paramNames, context } = info;
+    let RoutInfoImpl = class RoutInfoImpl {
```

Comment:
Do we actually need a class here? Seems like we could just use a POJO...

Reply:
If I do `Object.assign` is it going to get re-written to something in Ember?